spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java

```diff
@@ -470,10 +470,12 @@ public void testMulti() throws Exception {
 		assertThat(this.multiListener.latch2.await(60, TimeUnit.SECONDS)).isTrue();
 		ConsumerRecord<Integer, String> reply = KafkaTestUtils.getSingleRecord(consumer, "annotated8reply");
 		assertThat(reply.value()).isEqualTo("OK");
-		consumer.close();
 
 		template.send("annotated8", 0, 1, "junk");
 		assertThat(this.multiListener.errorLatch.await(60, TimeUnit.SECONDS)).isTrue();
+		ConsumerRecord<Integer, String> reply2 = KafkaTestUtils.getSingleRecord(consumer, "annotated8reply");
+		consumer.close();
+		assertThat(reply2.value()).isEqualTo("JUNK intentional");
```

Comment:
Did you change something in the logic I have missed that thrown exception is returned as normal reply even if that was `void` method?
Plus how that happened that `junk` was converted into upper case?